snap/types.go

```diff
@@ -139,9 +139,10 @@ func (confinementType *ConfinementType) fromString(str string) error {
 type ServiceStopReason string
 
 const (
-	StopReasonRefresh ServiceStopReason = "refresh"
-	StopReasonRemove  ServiceStopReason = "remove"
-	StopReasonDisable ServiceStopReason = "disable"
+	StopReasonRefresh    ServiceStopReason = "refresh"
+	StopReasonRemove     ServiceStopReason = "remove"
+	StopReasonDisable    ServiceStopReason = "disable"
+	StopReasonUserAction ServiceStopReason = "user"
```

Comment:
we need to think a bit here,  all the other reasons are verbs,  so this is really something like "stop"?  or "explicit-stop" ?

Reply:
Right. How about "user action"? Nb, I'd also introduce "hook action" (or similar) later.

Reply:
I dug a bit more, I see why this exist but is a bit strange, are you going to have later code that will check for these new values in wrappers/services.go and behave differently?

Reply:
No, my idea was to only log it (via existing Logf statements in wrappers).

Reply:
as discussed for now we probably just want StopReasonOther = "" and then not to log it